### PR TITLE
Caption support

### PR DESCRIPTION
**Pull request by artem-krachulov-paysera**
Attached optional caption property in the event info dictionary: `sendImage` and `sendVideo` methods

### CODE REVIEW

File: MatrixSDK/Data/MXRoom.m [hunk 1]
```diff
@@ -857,17 +858,35 @@ - (MXHTTPOperation*)sendImage:(NSData*)imageData
...
+    {
+        // Send message with caption
+        msgContent = [@{
+            @"msgtype": kMXMessageTypeImage,
+            @"body": filename,
+            @"caption": caption,
+            @"url": fakeMediaURI,
+            @"info": msgInfo
+        } mutableCopy];
+    }
```
Comment (ferologics) [inline MatrixSDK/Data/MXRoom.m]: why not just?
```suggestion
    if caption {
        msgContent[@"caption"] = msgContent
    }
```

File: MatrixSDK/Data/MXRoom.m [hunk 2]
```diff
@@ -825,6 +825,7 @@ - (MXHTTPOperation*)sendImage:(NSData*)imageData
 #elif TARGET_OS_OSX
                  andThumbnail:(NSImage*)thumbnail
 #endif
+                      caption:(NSString*)caption
```
Comment (ferologics) [inline MatrixSDK/Data/MXRoom.m:828]: maybe use `nullable`?

**Review by ferologics — CHANGES_REQUESTED** [review]
Can you add nullability to obj-c `caption` function parameters and simplify the logic for setting caption on `msgContent` dictionary?

File: MatrixSDK/Data/MXRoom.m (hunk 1 above)
Comment (artem-krachulov-paysera) [reply MatrixSDK/Data/MXRoom.m]: optimised

**Comment by danpe** [comment]
Guys, in general, why is this a fork from [xinsight/matrix-ios-sdk](https://github.com/xinsight/matrix-ios-sdk) and not [matrix-org/matrix-ios-sdk](https://github.com/matrix-org/matrix-ios-sdk) ?

File: MatrixSDK/Data/MXRoom.m (hunk 2 above)
Comment (artem-krachulov-paysera) [reply MatrixSDK/Data/MXRoom.m:828]: will work fine without `nullable` keyword, same as previous methods.

**Comment by ferologics** [comment]
@danpe because of the missing Carthage support for MatrixSDK.

**Review by ferologics — APPROVED** [review]

**Comment by ferologics** [comment]
@danpe the fork base has been updated to matrix-org/matrix-ios-sdk :)

**Pull request merged**